src/UrlMatcher.php

```diff
@@ -231,10 +232,24 @@ private function injectRoutes(): void
             if (!$route->getData('hasMiddlewares')) {
                 continue;
             }
-            $hostPattern = $route->getData('host') ?? '{_host:[a-zA-Z0-9\.\-]*}';
-            $methods = $route->getData('methods');
+
+            if ($route->isMultiHost()) {
```

Comment:
We can probably just get hosts and check if there are more than one in the resulting array because for end user the question about `isMultiHost` isn't really interesting.